In [ ]:
# Cinccinnati Paint Company

In [2]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score


In [3]:
data = pd.read_csv("CincinnatiPaintCompany.csv")
data.head()

Commissions  Calls  Driven
0           22    139    2371
1           13    132    2226
2           33    144    2731
3           38    142    3351
4           23    142    2289

1. Determinar las variables independientes


In [4]:
X = data[['Calls', 'Driven']]
X 

Calls  Driven
0     139    2371
1     132    2226
2     144    2731
3     142    3351
4     142    2289
5     142    3449
6     138    3114
7     139    3342
8     144    2842
9     134    2625
10    135    2121
11    137    2219
12    146    3463
13    146    3290
14    144    3103
15    147    2122
16    144    2791
17    149    3209
18    131    2287
19    144    2848
20    132    2690
21    132    2933
22    127    2671
23    154    2988
24    147    2829

# #2. Determinar la variable dependiente


In [5]:

y = data[['Commissions']]
y

Commissions
0            22
1            13
2            33
3            38
4            23
5            47
6            29
7            38
8            41
9            32
10           20
11           13
12           47
13           38
14           44
15           29
16           38
17           37
18           14
19           34
20           25
21           27
22           25
23           43
24           34

In [6]:
# Ajustar el modelo de regresión múltiple
model = LinearRegression()
model.fit(X, y) #Entrenando el modelo
y_pred = model.predict(X) #Predicciones del modelo
y_pred

#Métricas estadísticas
mse = mean_squared_error(y, y_pred) # Error de la media cuadrática
r2 = r2_score(y, y_pred) # Varianza

In [7]:
#3. valor de intersección
model.intercept_

array([-101.31444968])

In [8]:
# Coeficientes de la regresión
print("Coeficientes: ", model.coef_)
print("Error de la media cuadrática (MSE): ", mse)
print("Varianza (R^2): ", r2)

Coeficientes:  [[0.63233256 0.01568925]]
Error de la media cuadrática (MSE):  16.313412727936036
Varianza (R^2):  0.8352686374291527


In [9]:
# 4. Estimación del Rendimiento del Trabajador
Y = model.predict(np.array([[2800, 143]]))
print("Rendimiento estimado del trabajador: ", Y)

Rendimiento estimado del trabajador:  [[1671.46027242]]


/opt/anaconda3/lib/python3.13/site-packages/sklearn/utils/validation.py:2691: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Conclusión

Los resultados indican que los agentes que realizan más llamadas de ventas y recorren más millas tienden a obtener mayores comisiones. Ambos coeficientes son positivos, lo que muestra que tanto el esfuerzo en contacto con clientes como el desplazamiento contribuyen al aumento de las comisiones, y el valor de R a la 2 = 0.835, confirma que el modelo explica una proporción muy alta de la variación en los ingresos por comisiones. 